### PR TITLE
[https://nvbugs/6426834][fix] Deflake test_kv_transfer: cap NIXL progress threads, pin UCX env

### PR DESCRIPTION
**Pull request by chuangz0**
test_transfer_worker_v2[tp4_pp1_to_tp2_pp2] (and other multi-worker cases) intermittently hit the 120s per-test timeout in CI. Root cause: each NIXL transfer agent spawns TRTLLM_NIXL_NUM_THREADS (default 8) busy-polling UCX progress threads, and a single case builds up to 8 TransferWorkers (64 spinning threads per pytest process). On CI nodes shared by up to 8 single-GPU Jenkins agents the resulting CPU oversubscription inflates agent construction from ~3s to ~27s each, so 8 agents alone exceed the 120s timeout depending on neighbor load.

Measured on an H100 node with 4 concurrent pytest processes: 8 agents take ~178s with the default 8 progress threads vs ~23s with 1 thread.

Fix, applied to all unit tests that create real UCX/NIXL transfer agents (test_kv_transfer.py, test_kv_transfer_mp.py, test_agent.py, test_agent_multi_backends.py, test_cache_transceiver_single_process.py, test_py_cache_transceiver_mp.py, test_deepseek_v4_kv_transfer.py, test_mamba_transfer.py, others/test_kv_cache_transceiver.py):
- Set TRTLLM_NIXL_NUM_THREADS=1 with an explicit assignment (not setdefault, so an externally supplied value cannot defeat the cap): these tests verify transfer logic, not transfer-engine threading (sender-side KV_TRANSFER_NUM_THREADS worker threads are unchanged).
- Force UCX_TLS=^ib,gdr_copy with an explicit assignment instead of the previous setdefault, which was inert in CI (the CI agent bootstrap exports UCX_TLS before pytest starts).

Un-waive the affected case.

Validation: 8 concurrent whole-file runs x 5 iterations (40 runs) on one H100 node under CI-parity env all passed; before the fix the same setup timed out in the first iteration on every GPU.

## Dev Engineer Review

- Standardizes UCX/NIXL test configuration across KV-transfer and transceiver tests:
  - Forces `UCX_TLS=^ib,gdr_copy`.
  - Limits `TRTLLM_NIXL_NUM_THREADS` to `1` to prevent CPU oversubscription.
- Configuration changes are consistent and scoped to tests using real UCX/NIXL agents.
- The affected KV-transfer test waiver was removed; the unrelated proxy shutdown test was added to `waives.txt` with valid formatting.
- No public APIs or test logic were changed. CI completed successfully, including 40 whole-file runs on an H100 node.

## QA Engineer Review

- Modified test modules configure runtime environments; no test functions were added, modified, or removed.
- The previously waived `unittest/disaggregated/test_kv_transfer.py::test_transfer_worker_v2[tp4_pp1_to_tp2_pp2]` is now un-waived.
- Test coverage is validated by the successful CI pipeline and 40 repeated whole-file runs.
- **Verdict: sufficient**

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- If PR introduces API changes, an appropriate PR label is added - either `api-compatible` or `api-breaking`. For `api-breaking`, include `BREAKING` in the PR title.
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`.

### CODE REVIEW

File: tests/unittest/disaggregated/test_kv_transfer.py [hunk 1]
```diff
@@ -5,8 +5,24 @@
...
+os.environ["UCX_TLS"] = "^ib,gdr_copy"
+os.environ.pop("UCX_NET_DEVICES", None)
+# Each NIXL agent spawns TRTLLM_NIXL_NUM_THREADS (default 8) busy-polling
+# progress threads, and a single case builds up to 8 TransferWorkers (one per
+# rank). On CI nodes shared with other single-GPU jobs the resulting CPU
+# oversubscription inflates agent construction from ~3s to ~30s each, blowing
+# the 120s per-test timeout intermittently (https://nvbugs/6426834). One
+# progress thread is enough here: these tests verify transfer logic, not
+# transfer-engine threading.
+os.environ.setdefault("TRTLLM_NIXL_NUM_THREADS", "1")
```
[the remark this answers is not in the document]
Comment (chuangz0) [reply tests/unittest/disaggregated/test_kv_transfer.py]: Fixed in 20a8846: `TRTLLM_NIXL_NUM_THREADS` is now assigned explicitly (no `setdefault`) in both `test_kv_transfer.py` and `test_kv_transfer_mp.py`. The same deterministic UCX/NIXL env pinning (`UCX_TLS=^ib,gdr_copy`, drop `UCX_NET_DEVICES`, `TRTLLM_NIXL_NUM_THREADS=1`) is also applied to the other unit tests that create real transfer agents: `test_agent.py`, `test_agent_multi_backends.py`, `test_cache_transceiver_single_process.py`, `test_py_cache_transceiver_mp.py`, `test_deepseek_v4_kv_transfer.py`, `test_mamba_transfer.py`, and `others/test_kv_cache_transceiver.py`.

File: tests/integration/test_lists/waives.txt [hunk 2]
```diff
@@ -436,7 +436,6 @@ unittest/_torch/visual_gen/multi_gpu/test_ulysses_async.py::test_capture_smoke S
 unittest/_torch/visual_gen/test_attention_integration.py::test_sage_attention_self_attention[fp8-2-1560] SKIP (https://nvbugs/6198760)
 unittest/_torch/visual_gen/test_attention_integration.py::test_sage_attention_self_attention[int8-2-1560] SKIP (https://nvbugs/6198760)
 unittest/auto_deploy/multigpu/custom_ops SKIP (https://nvbugs/6403920)
-unittest/disaggregated/test_kv_transfer.py::test_transfer_worker_v2[tp4_pp1_to_tp2_pp2] SKIP (https://nvbugs/6426834)
```
Comment (Shixiaowei02) [inline tests/integration/test_lists/waives.txt:368]: Thanks for the fix! Please also check the related bugs for any waived cases.
https://nvbugs/6450092 , https://nvbugs/6403793 , https://nvbugs/6445316

**Review by Shixiaowei02 — APPROVED** [review]

File: tests/integration/test_lists/waives.txt [hunk 3]
```diff
@@ -363,9 +363,7 @@ unittest/_torch/visual_gen/multi_gpu/test_ulysses_async.py::test_capture_smoke S
 unittest/_torch/visual_gen/test_attention_integration.py::test_sage_attention_self_attention[fp8-2-1560] SKIP (https://nvbugs/6198760)
 unittest/_torch/visual_gen/test_attention_integration.py::test_sage_attention_self_attention[int8-2-1560] SKIP (https://nvbugs/6198760)
 unittest/auto_deploy/multigpu/custom_ops SKIP (https://nvbugs/6403920)
-unittest/disaggregated/test_kv_transfer.py SKIP (https://nvbugs/6403793)
-unittest/disaggregated/test_kv_transfer.py::test_transfer_worker_v2[tp1_pp4_to_tp2_pp2] SKIP (https://nvbugs/6445316)
-unittest/disaggregated/test_kv_transfer.py::test_transfer_worker_v2[tp4_pp1_to_tp2_pp2] SKIP (https://nvbugs/6426834)
+unittest/executor/test_proxy_fast_death.py::test_pool_session_shutdown_never_blocks_after_release SKIP (https://nvbugs/6480574)
```
Comment (chzblych) [inline tests/integration/test_lists/waives.txt]: This issue was already resolved by https://github.com/NVIDIA/TensorRT-LLM/pull/16630?

File: tests/unittest/disaggregated/test_agent.py [hunk 4]
```diff
@@ -6,8 +6,10 @@
 import pytest
 import torch
 
-# Exclude IB (no fabric) and gdr_copy (UCX rcache SIGABRT at teardown).
-os.environ.setdefault("UCX_TLS", "^ib,gdr_copy")
+# Force a deterministic UCX/NIXL config regardless of what the cluster/CI
+# injects; see test_kv_transfer.py for the full rationale.
+os.environ["UCX_TLS"] = "^ib,gdr_copy"
```
Comment (chzblych) [inline tests/unittest/disaggregated/test_agent.py:11]: Does this mean we should also set `UCX_TLS=^ib,gdr_copy` in our CI environment? If so, please file a ticket and let me know. We can address it later.

**Review by chzblych — APPROVED** [review]

File: tests/unittest/disaggregated/test_agent.py (hunk 4 above)
Comment (chuangz0) [reply tests/unittest/disaggregated/test_agent.py:11]: These tests require only one GPU and do not require network support; this environment variable is set to prevent complex network environments from affecting  data transfer within one GPU. In multi-node setup, different clusters may require different environment variables, so setting  fixed environment variable in the CI environment is not recommended.

File: tests/integration/test_lists/waives.txt (hunk 3 above)
Comment (chuangz0) [reply tests/integration/test_lists/waives.txt]: Good catch — this waive was accidentally re-added during a rebase; the issue was indeed already fixed by #16630. Removed in 127e7f4b0b.

**Comment by chuangz0** [comment]
/bot skip --comment "all tests have passed"
